qute.jdt/com.redhat.qute.jdt/src/main/java/com/redhat/qute/jdt/internal/template/datamodel/TemplateDataSupport.java

```diff
@@ -0,0 +1,161 @@
+/*******************************************************************************
+* Copyright (c) 2022 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.qute.jdt.internal.template.datamodel;
+
+import static com.redhat.qute.jdt.internal.QuteJavaConstants.TEMPLATE_DATA_ANNOTATION;
+import static com.redhat.qute.jdt.internal.QuteJavaConstants.TEMPLATE_EXTENSION_ANNOTATION_MATCH_NAME;
+import static com.redhat.qute.jdt.internal.QuteJavaConstants.TEMPLATE_EXTENSION_ANNOTATION_NAMESPACE;
+
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.IAnnotatable;
+import org.eclipse.jdt.core.IAnnotation;
+import org.eclipse.jdt.core.IField;
+import org.eclipse.jdt.core.IJavaElement;
+import org.eclipse.jdt.core.IMethod;
+import org.eclipse.jdt.core.IType;
+import org.eclipse.jdt.core.JavaModelException;
+
+import com.redhat.qute.commons.datamodel.resolvers.ValueResolverInfo;
+import com.redhat.qute.jdt.QuteSupportForTemplate;
+import com.redhat.qute.jdt.internal.resolver.ITypeResolver;
+import com.redhat.qute.jdt.template.datamodel.AbstractAnnotationTypeReferenceDataModelProvider;
+import com.redhat.qute.jdt.template.datamodel.SearchContext;
+import com.redhat.qute.jdt.utils.AnnotationUtils;
+
+/**
+ * Template data support.
+ *
+ * <code>
+ * &#64;TemplateData
+ * class Item {
+ *
+ * 		public final BigDecimal price;
+ *
+ * 		public Item(BigDecimal price) {
+ * 			this.price = price;
+ * 		}
+ *
+ * 		public BigDecimal getDiscountedPrice() {
+ * 			return price.multiply(new BigDecimal("0.9"));
+ * 		}
+ * }
+ * </code>
+ *
+ *
+ * @see https://quarkus.io/guides/qute-reference#template_data
+ *
+ */
+public class TemplateDataSupport extends AbstractAnnotationTypeReferenceDataModelProvider {
+
+	private static final Logger LOGGER = Logger.getLogger(TemplateDataSupport.class.getName());
+
+	private static final String[] ANNOTATION_NAMES = {
+			TEMPLATE_DATA_ANNOTATION
+
+	};
+
+	@Override
+	protected String[] getAnnotationNames() {
+		return ANNOTATION_NAMES;
+	}
+
+	@Override
+	protected void processAnnotation(IJavaElement javaElement, IAnnotation annotation, String annotationName,
+			SearchContext context, IProgressMonitor monitor) throws JavaModelException {
+		if (!(javaElement instanceof IAnnotatable)) {
+			return;
+		}
+		IAnnotation templateData = AnnotationUtils.getAnnotation((IAnnotatable) javaElement, TEMPLATE_DATA_ANNOTATION);

```

Comment:
Please note that you can annotate a Jav atype class with several @TemplateData, so you need to loop for each annotations and check if the current annotation is @TemplateData. The main idea is to merge teh attribute value (take the first namespace that you find)

With your current code if you write:

```java
@TemplateData
@TemplateData(namespace = "FOO")
@TemplateData(namespace = "BAR")
public class Item
```

you will get namespace = null, although you should have namespace = "FOO".